app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -211,22 +211,57 @@ def update(self, experiment, data):
 class NimbusProbeSetUpdateSerializer(
     NimbusChangeLogMixin, NimbusStatusRestrictionMixin, serializers.ModelSerializer
 ):
-    probe_sets = serializers.PrimaryKeyRelatedField(
+    primary_probe_sets = serializers.PrimaryKeyRelatedField(
+        many=True, queryset=NimbusProbeSet.objects.all()
+    )
+
+    secondary_probe_sets = serializers.PrimaryKeyRelatedField(
         many=True, queryset=NimbusProbeSet.objects.all()
     )
 
     class Meta:
         model = NimbusExperiment
-        fields = ("probe_sets",)
+        fields = ("primary_probe_sets", "secondary_probe_sets")
+
+    def validate(self, data):
+        """Validate the probe sets don't overlap and have no more than 2 primary.
+
+        Note that the default DRF validation ensures all the probe id's are valid and
+        it will not save overlapping probesets. It does not however throw an error with
+        overlapping probesets, so that is checked explicitly here.
+
+        """
+        data = super().validate(data)
+        primary_probe_sets = set(data["primary_probe_sets"])
+        secondary_probe_sets = set(data["secondary_probe_sets"])
+        if primary_probe_sets.intersection(secondary_probe_sets):
+            raise serializers.ValidationError(
+                {
+                    "primary_probe_sets": (
+                        "Primary probe sets cannot overlap with secondary probe sets."
+                    )
+                }
+            )
+        if len(primary_probe_sets) > 2:
+            raise serializers.ValidationError(
+                {"primary_probe_sets": ("Exceeded maximum primary probe set limit of 2.")}
+            )
```

Comment:
Usually per field validation can go in a separate method, only cross field validation needs to go in `validate` so this could go in `def validate_primary_probe_sets`

Reply:
Ah right.